### DISCUSSION TITLE
Announcement

### DISCUSSION THREAD

**Discussion by adrianomic** [mastodon/mastodon#17035] [Q&A] [answered]
What does Announcement status_ids field do?
I don't see this field is used in front-end side.

```
You can use post syntax. Please be mindful of the space the announcement will take up on the user's screen
```

Here, what do you mean by post syntax?

**Comment by ClearlyClaire** [comment]
> What does Announcement status_ids field do?
> I don't see this field is used in front-end side.

It records statuses that were mentioned (linked to) in the announcement. In the REST API, it's `statuses`, and it's used by the WebUI in `app/javascript/mastodon/features/getting_started/components/announcements.js` to open linked statuses in the WebUI instead of opening a new tab.

> Here, what do you mean by post syntax?

I assume this means hashtags, custom emojis and mentions.

**Reply by adrianomic** [reply]
How to mention status in announcement? Can you please give me more details?

**Reply by ClearlyClaire** [reply]
Post the link to a known status in the announcement text. When clicked from the WebUI, this will open the status in the WebUI rather than opening a new tab.

**Reply by adrianomic** [reply]
Ok, thanks

**Reply by adrianomic** [reply]
I put status link in announcement. But it does not update ``status_ids`` field.
Do we have other way to link status to announcement?

**Reply by ClearlyClaire** [reply]
What does the link look like? Are you using 3.4.3 or main? It works here.

**Reply by adrianomic** [reply]
I use 3.4.1
If I put link and the link is like https://website.com/@username/status_id
Just wondering if it parse text and extract links from the text.

**Comment by ClearlyClaire** [comment]
3.4.1 has a bug that makes it fails to handle links in announcements properly

**Reply by adrianomic** [reply]
Do you have PR for this fix?

**Reply by adrianomic** [reply]
https://github.com/mastodon/mastodon/compare/v3.4.2...v3.4.3

v3.4.2 and v3.4.3 few commits
v3.4.2 and v3.4.1 same

**Comment by ClearlyClaire** [comment, marked as answer]
It is fixed in main and 3.4.3.

The PR is https://github.com/mastodon/mastodon/pull/16941

**Comment by adrianomic** [comment]
Thanks @ClearlyClaire , it works

**Reply by adrianomic** [reply]
It does not recognize localhost:3000 url. How can I do this shortly?
